server/odc-service/src/main/java/com/oceanbase/odc/service/monitor/MeterHolder.java

```diff
@@ -0,0 +1,363 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.service.monitor;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Supplier;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+import com.oceanbase.odc.service.monitor.MonitorAutoConfiguration.BusinessMeterRegistry;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.Counter.Builder;
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Timer;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.Getter;
+import lombok.Value;
+import lombok.extern.slf4j.Slf4j;
+
+@Component
+@Slf4j
+@Getter
+@ConditionalOnProperty(value = "odc.system.monitor.actuator.enabled", havingValue = "true")
+public class MeterHolder implements MeterClear {
+
+    private final GaugeHolder gaugeHolder;
+    private final CounterHolder counterHolder;
+    private final TimerHolder timerHolder;
+
+    public MeterHolder(BusinessMeterRegistry meterRegistry,
```

Comment:
introduce registerMeter and unregisterMeter to avoid use if null operation.
any regular operation after registerMeter find null value should give some warn

Reply:
removed